api/instance/v1/instance_utils.go

```diff
@@ -83,15 +95,35 @@ type UpdateIPRequest struct {
 	Zone    scw.Zone             `json:"-"`
 	IP      string               `json:"-"`
 	Reverse *NullableStringValue `json:"reverse"`
+	Server  *string              `json:"server"`
 }
 
 // UpdateIP updates an IP
+// We need to transform UpdateIPRequest into updateIPRequest because updateIPRequest is not usable
+// as a public interface.
 func (s *API) UpdateIP(req *UpdateIPRequest, opts ...scw.RequestOption) (*UpdateIPResponse, error) {
-	ipResponse, err := s.updateIP(&updateIPRequest{
+
+	requestPrivate := &updateIPRequest{
 		Zone:    req.Zone,
 		IP:      req.IP,
 		Reverse: req.Reverse,
-	})
+	}
+
+	switch {
+	case req.Server == nil:
+		// Do nothing
+
+	case *req.Server == "":
+		// Setup server field to detach old server
```

Comment:
Are we sure we want to use an empty string to detach a server? It was one of the reasons for which we have created `DetachIP`.
If we want to keep this behavior (`""` == detach) we should document it in the method godoc.

Reply:
We use the same pattern for reverse already.

Reply:
Reverse use a `*NullableStringValue`. We should use the same implementation for `Reverse` and `Server` and document it in the method godoc.

Reply:
If we use the same pattern, why do we have
```
	Reverse *NullableStringValue `json:"reverse"`
	Server  *string              `json:"server"`
```
Do we need to change Reverse ?
Do we need to change all `*NullableStringValue` (cf internal ticket APIGW-1721) ?